### ISSUE TITLE
from npm to pnpm

### ISSUE THREAD

**Issue by xiejiaxin007** [pnpm/pnpm#3347]
i want update my project,from npm to pnpm,but when i change,it have a problem like this:

This dependency was not found:
* regenerator-runtime/runtime.js in ./src/assets/js/upload/upload.js, ./src/assets/js/utils/utils.js and 1 other
To install it, you can run: npm install --save regenerator-runtime/runtime.js
Failed to resolve loader: cache-loader
You may need to install it.
Failed to resolve loader: vue-style-loader
You may need to install it.

pnpm: 6.0.2
node:v10.15.3
vuecli4

**Comment by zkochan** [comment]
https://pnpm.io/faq#solution-3

**Comment by xiejiaxin007** [comment]
I didn't understand what you sent. I tried to use.npmrc file for configuration, but it didn't work

**Comment by xiejiaxin007** [comment]
sorry，I didn't understand what you sent. I tried to use.npmrc file for configuration, but it didn't work

**Comment by xiejiaxin007** [comment]
I have tried the scheme you mentioned, but I only need to delete node_modules once, and it will report an error. Is there any way?

**Comment by xiejiaxin007** [comment]
In addition, I see that the solution you sent does not suggest that I directly use PNPM install --shamefully fully hoist, is there a better way？

**Comment by zkochan** [comment]
Yes, you can set `shamefully-hoist=true` in an `.npmrc` file in the root of your project.

**Comment by zkochan** [comment]
Closing: current pnpm covers this.

A vue-cli 4 project whose loaders were not declared as dependencies. Declare them, or hoist with `publicHoistPattern`.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
